kai/reactive_codeplanner/task_manager/task_manager.py

```diff
@@ -269,24 +269,35 @@ def handle_new_tasks_after_processing(self, task: Task) -> None:
 
     def should_skip_task(self, task: Task) -> bool:
         skip = task in self.processed_tasks or task in self.ignored_tasks
-        logger.debug("Should skip task %s: %s", task, skip)
+        logger.log(logging.TRACE, "Should skip task %s: %s", task, skip)
         return skip
 
     def is_similar_to_task(self, task1: Task, task2: Optional[Task]) -> bool:
         if task2 is None:
             return False
         same = task2 == task1
         if same:
-            logger.debug("Task %s is same to prior task %s: %s", task1, task2, same)
+            logger.log(
+                logging.TRACE,
+                "Task %s is same to prior task %s: %s",
+                task1,
+                task2,
+                same,
+            )
+            return same
         # TODO(fabianvf): Give tasks the ability to provide a specific fuzzy equals function?
         similar = False
         if hasattr(task1, "fuzzy_equals"):
             similar = task1.fuzzy_equals(task2, offset=2)
             if similar:
-                logger.debug(
-                    "Task %s is similar to prior task %s: %s", task1, task2, similar
+                logger.log(
+                    logging.TRACE,
+                    "Task %s is similar to prior task %s: %s",
+                    task1,
+                    task2,
+                    similar,
                 )
-        return same or similar
+        return False
```

Comment:
I don't think this change was intentional? This is probably what broke the CI too